### PR TITLE
Add access_tier attribute to storage_share 

### PR DESCRIPTION
**Pull request by koikonom**

### CODE REVIEW

File: internal/services/storage/storage_share_resource.go [hunk 1]
```diff
@@ -222,6 +236,10 @@ func resourceStorageShareRead(d *pluginsdk.ResourceData, meta interface{}) error
 	d.Set("url", id.ID())
 	d.Set("enabled_protocol", string(props.EnabledProtocol))
 
+	if accessTier := props.AccessTier; accessTier != nil {
+		d.Set("access_tier", string(*accessTier))
+	}
```
Comment (tombuildsstuff) [inline internal/services/storage/storage_share_resource.go:241]: this'll want a value always set:

```suggestion
	accessTier := ""
	if props.AccessTier != nil {
		accessTier = string(*props.AccessTier)
	}
	d.Set("access_tier", accessTier)
```

File: internal/services/storage/storage_share_resource_test.go [hunk 2]
```diff
@@ -185,6 +186,35 @@ func TestAccStorageShare_largeQuota(t *testing.T) {
...
+		data.ImportStep(),
+		{
+			Config: r.accessTier(data, string(shares.HotAccessTier)),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.accessTier(data, string(shares.TransactionOptimizedAccessTier)),
```
Comment (tombuildsstuff) [inline internal/services/storage/storage_share_resource_test.go:209]: can we hard-code the string values for these? that protects against the value changing casing etc

File: website/docs/d/storage_share.html.markdown [hunk 3]
```diff
@@ -37,6 +37,8 @@ The following arguments are supported:
 
 * `quota` - The quota of the File Share in GB.
 
+* `access_tier` - The tier of the File Share. Can be one of `Hot`, `Cool`, `TransactionOptimized`, or `Premium`.
```
Comment (tombuildsstuff) [inline website/docs/d/storage_share.html.markdown:40]: Premium isn't a supported value? https://github.com/hashicorp/terraform-provider-azurerm/pull/16462/files#diff-773e6c18c4552e9b5d5496623bb06e64cc9703e228698492c8f33be90d700781R128-R130

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
Left a few comments inline but this otherwise LGTM 👍

**Review by tombuildsstuff — APPROVED** [review]
LGTM 👍

**Review by wernerfred — COMMENTED** [review]
@tombuildsstuff @koikonom shouldn't the `website/docs/d/storage_share.html.markdown` (datasource) go into `website/docs/r/storage_share.html.markdown` (resources)?

**Pull request merged**